src/core/models.py

```diff
@@ -0,0 +1,283 @@
+        """The id of the unit from the unit name.
+
+        e.g opensearch-dashboards/2 --> 2
+        """
+        return int(self.component.name.split("/")[1])
+
+    # -- Cluster Init --
+
+    @property
+    def started(self) -> bool:
+        """Flag to check if the unit has started the service."""
+        return self.relation_data.get("state", None) == "started"
+
+    @property
+    def password_rotated(self) -> bool:
+        """Flag to check if the unit has rotated their internal passwords."""
+        return bool(self.relation_data.get("password-rotated", None))
+
+    @property
+    def hostname(self) -> str:
+        """The hostname for the unit."""
+        return socket.gethostname()
+
+    @property
+    def fqdn(self) -> str:
+        """The Fully Qualified Domain Name for the unit."""
+        return socket.getfqdn()
+
+    @property
+    def private_ip(self) -> str:
+        """The IP for the unit."""
+        return socket.gethostbyname(self.hostname)
+
+    @property
+    def public_ip(self) -> str:
+        result = subprocess.check_output(
+            [
+                "bash",
+                "-c",
+                "ip a | grep global | grep -v 'inet 10.' | cut -d' ' -f6 | cut -d'/' -f1",
+            ],
+            text=True,
+        )
+        return result.rstrip()
+
+    @property
+    def host(self) -> str:
+        """The hostname for the unit."""
+        host = ""
+        if self.substrate == "vm":
+            for key in ["hostname", "ip", "private-address"]:
+                if host := self.relation_data.get(key, ""):
+                    break
+
+        if self.substrate == "k8s":
+            host = f"{self.component.name.split('/')[0]}-{self.unit_id}.{self.component.name.split('/')[0]}-endpoints"
+
+        return host
+
+    # -- TLS --
+
+    @property
+    def private_key(self) -> str:
+        """The private-key contents for the unit to use for TLS."""
+        return self.relation_data.get("private-key", "")
+
+    # @property
+    # def keystore_password(self) -> str:
+    #     """The Java Keystore password for the unit to use for TLS."""
+    #     return self.relation_data.get("keystore-password", "")
+    #
+    # @property
+    # def truststore_password(self) -> str:
+    #     """The Java Truststore password for the unit to use for TLS."""
+    #     return self.relation_data.get("truststore-password", "")
+
+    @property
+    def csr(self) -> str:
+        """The current certificate signing request contents for the unit."""
+        return self.relation_data.get("csr", "")
+
+    @property
+    def certificate(self) -> str:
+        """The certificate contents for the unit to use for TLS."""
+        return self.relation_data.get("certificate", "")
+
+    @property
+    def ca(self) -> str:
+        """The root CA contents for the unit to use for TLS."""
+        return self.relation_data.get("ca-cert", "")
+
+    @property
+    def sans(self) -> dict[str, list[str]]:
```

Comment:
```suggestion
    def sans(self) -> Dict[str, List[str]]:
```

Reply:
See https://github.com/canonical/opensearch-dashboards-operator/pull/1#discussion_r1557363244, I'd prefer NOT to do this.